src/main/java/org/openpnp/machine/reference/ReferenceNozzle.java

```diff
@@ -264,7 +254,7 @@ public void moveToSafeZ(double speed) throws Exception {
     @Override
     public void home() throws Exception {
         Logger.debug("{}.home()", getName());
-        for (NozzleTip attachedNozzleTip : this.getNozzleTips()) {
+        for (NozzleTip attachedNozzleTip : this.getCompatibleNozzleTips()) {
```

Comment:
Same here, as above. This will potentially cause the code below to run multiple times per nozzle tip, since multiple nozzles may have the nozzle tip configured as compatible. Perhaps it would make sense to do this on Machine.home() since that is where NozzleTips live now?

Reply:
Again correct with the additions I made.